src/hotspot/cpu/x86/methodHandles_x86.cpp

```diff
@@ -555,18 +555,19 @@ void trace_method_handle_stub(const char* adaptername,
       PreserveExceptionMark pem(Thread::current());
       FrameValues values;
 
-      // Current C frame
-      frame cur_frame = os::current_frame();
+      frame cur_frame = os::current_frame(); // reports caller's frame!
```

Comment:
It is unclear what "caller" is relative to in that comment - is it the caller of os::current_frame() as seen by os::current_frame(), or is it the caller of the current method that is invoking os::current_frame() ?

Reply:
As opposed to "current C frame", It is the caller of the "current C frame" ;-)

I'll think about a way to make the comment less ambiguous.

Reply:
Why I thought that explicitly calling that it's the "caller's frame" makes things clearer: 
  (1) the code explicitly is written in such a way to skip any intermediate frames;
  (2) all users are in generated code.

So, we know for sure there's a separate frame on the stack (which contains register values) and it differs from the top one. The bug is that we can't skip the frame returned by `os::current_frame()` because it doesn't return the top frame on the stack, but the next one.

Reply:
Removing the comment doesn't help the reader understand the code.
Please replace the comment with something that accurately explains
what happens from this `os::current_frame()` call site.

Reply:
Done. Is it better now?